diff --git a/build/make/configure.sh b/build/make/configure.sh
index b5151da..a739468 100755
--- a/build/make/configure.sh
+++ b/build/make/configure.sh
@@ -661,9 +661,14 @@ process_common_toolchain() {
             x86_64*mingw32*)
                 tgt_os=win64
                 ;;
-            *mingw32*|*cygwin*)
+            *mingw32*|*cygwin*|*win32-gcc*)
                 [ -z "$tgt_isa" ] && tgt_isa=x86
                 tgt_os=win32
+                [ "$tgt_isa" == "x86_64" ] && tgt_os=win64
+                ;;
+            *win64-gcc*)
+                [ -z "$tgt_isa" ] && tgt_isa=x86_64
+                tgt_os=win64
                 ;;
             *linux*|*bsd*)
                 tgt_os=linux
diff --git a/configure b/configure
index 9f5a435..93f3d30 100755
--- a/configure
+++ b/configure
@@ -137,6 +137,7 @@ all_platforms="${all_platforms} x86_64-darwin13-gcc"
 all_platforms="${all_platforms} x86_64-linux-gcc"
 all_platforms="${all_platforms} x86_64-linux-icc"
 all_platforms="${all_platforms} x86_64-solaris-gcc"
+all_platforms="${all_platforms} x86_64-win32-gcc"
 all_platforms="${all_platforms} x86_64-win64-gcc"
 all_platforms="${all_platforms} x86_64-win64-vs8"
 all_platforms="${all_platforms} x86_64-win64-vs9"
